common/src/Microsoft.Azure.IIoT.Hub.Module.Client/src/Default/IoTSdkFactory.cs

```diff
@@ -106,12 +106,11 @@ public sealed class IoTSdkFactory : IClientFactory, IDisposable {
             }
             if (!string.IsNullOrEmpty(ehubHost)) {
                 // Running in edge mode 
-                // the configured transport will be forced to it's OverTcp variant as follows:
-                // MqttOverTcp when Mqtt, MqttOverWebsocket or MqttOverTcp 
-                // and AmqpOverTcp otherwise
-
-                if (config.Transport != TransportOption.Any &&
-                    (config.Transport & TransportOption.Mqtt) != 0) {
+                // the configured transport (if provided) will be forced to it's OverTcp
+                // variant as follows: AmqpOverTcp when Amqp, AmqpOverWebsocket or AmqpOverTcp specified
+                // and MqttOverTcp otherwise. Default is MqttOverTcp
+                if ((config.Transport & TransportOption.Mqtt) != 0) {
```

Comment:
@marcschier @cristipogacean I didn't have time to review this, but did we opt for the bitwise op for performance reasons over config.Transport.HasFlag(TransportOption.Mqtt), which has better readability?